component-tests/cxx/src/test_containers.cpp

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2025 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#include "iox2/component-tests/common.hpp"
+
+#include <iox2/container/static_vector.hpp>
+
+#include <iostream>
+
+namespace {
+class ContainerTest : public IComponentTest {
+  public:
+    ContainerTest() = default;
+    ContainerTest(ContainerTest const&) = delete;
+    ContainerTest(ContainerTest&&) = delete;
+    auto operator=(ContainerTest const&) -> ContainerTest& = delete;
+    auto operator=(ContainerTest&&) -> ContainerTest& = delete;
+    ~ContainerTest() override = default;
+    auto test_name() const -> char const* override {
+        return "containers";
+    }
+    auto run_test(iox2::Node<iox2::ServiceType::Ipc> const& node) -> bool override;
+};
+
+struct ContainerTestRequest {
+    // IOX2_TYPE_NAME is equivalent to the payload type name used on the Rust side
+    static constexpr const char* IOX2_TYPE_NAME = "ContainerTestRequest";
+    int32_t vector_type_sequence;
```

Comment:
This could then become
```suggestion
    VectorTypeSequence vector_type_sequence;
```

Which would then also make the casts to `int` further down obsolete.